diff --git adapter-bitbang.c adapter-bitbang.c
index f51716d..0dd89ea 100644
--- ./adapter-bitbang.c
+++ ./adapter-bitbang.c
@@ -1200,6 +1200,9 @@ adapter_t *adapter_open_bitbang(const char *port, int baud_rate)
                             // "ascii ICSP v1X"
     serial_write(&ch, 1);
     n = serial_read(buffer, 14, 250);
+    fprintf(stderr, "\nn = %d (should be 14\n", n);
+    buffer[15] = 0;
+    fprintf(stderr, "\nbuffer = %s\n", buffer);
 
     if (n == 14 && memcmp(buffer, "ascii ICSP v1", 13) == 0)
         printf(" OK2 - %s\n", buffer);
